File: app/src/main/java/tomaszkopacz/meetbam/activities/MainApp.java
package tomaszkopacz.meetbam.activities;

import android.app.Application;

import tomaszkopacz.meetbam.di.components.DaggerWebServiceComponent;
import tomaszkopacz.meetbam.di.components.WebServiceComponent;

/**
 * Created by tomas on 19.02.2018.
 */

public class MainApp extends Application {

    private WebServiceComponent mWebServiceComponent;

    @Override
    public void onCreate() {
        super.onCreate();
        mWebServiceComponent = DaggerWebServiceComponent.builder().build();
    }

    /**
     * Returns web service component.
     * @return
     */
    public WebServiceComponent getWebServiceComponent(){
        return mWebServiceComponent;
    }
}

File: app/src/main/java/tomaszkopacz/meetbam/view/MainApp.kt
package tomaszkopacz.meetbam.view

import android.app.Application
import tomaszkopacz.meetbam.di.AppComponent
import tomaszkopacz.meetbam.di.DaggerAppComponent
import tomaszkopacz.meetbam.entity.User
import tomaszkopacz.meetbam.service.Preferences
import tomaszkopacz.meetbam.service.TypefaceUtil

class MainApp : Application() {

    var component: AppComponent? = null
        private set

    private var prefs: Preferences? = null

    override fun onCreate() {
        super.onCreate()

        component = DaggerAppComponent.builder().build()

        TypefaceUtil.overrideFont(
                this,
                "SERIF",
                "Comfortaa_Regular.ttf")

        prefs = Preferences(this)
    }

    fun login(mail: String, name: String, surname: String, password: String) {

        prefs!!.name = name
        prefs!!.surname = surname
        prefs!!.mail = mail
        prefs!!.password = password
    }

    fun logout() {
        prefs!!.removeAll()
    }

    fun isUserLoggedIn(): Boolean {
        return (prefs!!.name != prefs!!.NO_USER && prefs!!.surname != prefs!!.NO_USER &&
                prefs!!.mail != prefs!!.NO_USER && prefs!!.password != prefs!!.NO_USER)
    }

    fun getLoggedUser(): User {
        val user = User()

        if (isUserLoggedIn()) {
            user.name = prefs!!.name
            user.surname = prefs!!.surname
            user.mail = prefs!!.mail
            user.password = prefs!!.password
        }

        return user
    }
}
